src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2549,22 +2535,21 @@ public override BoundNode VisitObjectCreationExpression(BoundObjectCreationExpre
             return null;
         }
 
-        public override BoundNode VisitUnconvertedObjectCreationExpression(BoundUnconvertedObjectCreationExpression node)
+        public override BoundNode? VisitUnconvertedObjectCreationExpression(BoundUnconvertedObjectCreationExpression node)
         {
             var discardedDiagnostics = new DiagnosticBag();
-            var expr = _binder.BindObjectCreationForErrorRecovery(node, discardedDiagnostics);
+            var expr = _binder!.BindObjectCreationForErrorRecovery(node, discardedDiagnostics);
```

Comment:
Should be an assertion

Reply:
No need. The dereference will throw an exception.

---
In reply to: [463268923](https://github.com/dotnet/roslyn/pull/45640#discussion_r463268923) [](ancestors = 463268923)